compiler/noirc_frontend/src/ast/expression.rs

```diff
@@ -688,10 +688,12 @@ impl Display for FunctionDefinition {
 }
 
 impl FunctionReturnType {
-    pub fn get_type(&self) -> &UnresolvedTypeData {
+    pub fn get_type(&self) -> UnresolvedType {
         match self {
-            FunctionReturnType::Default(_span) => &UnresolvedTypeData::Unit,
-            FunctionReturnType::Ty(typ) => &typ.typ,
+            FunctionReturnType::Default(span) => {
+                UnresolvedType { typ: UnresolvedTypeData::Unit, span: Some(*span) }
+            }
+            FunctionReturnType::Ty(typ) => typ.clone(),
```

Comment:
I've added the cow, https://github.com/noir-lang/noir/pull/2652/commits/7c774dd78d74723df46efc41e7ee9d840b18dc6e

We don't gain that much from it, since in both places where it's used it's immediately passed to `resolve_type()` which takes `UnresolvedType` by value - maybe we should change it to take by ref.